### PR TITLE
Re-add support for itemtypes in CondIsOfType

### PR DESCRIPTION
**Pull request by Mr-Darth**
### Description
Looks like #1597 accidentally removed `itemtypes` support in CondIsOfType. This PR adds it back.

---
**Target Minecraft Versions:** 
**Requirements:** 
**Related Issues:**

### CODE REVIEW

**Review by AyhamAl-Ali — APPROVED** [review]

**Comment by TheLimeGlass** [comment]
There is no entitytype to entitydata converter, this will break comparing an entity to an entitytype.

**Comment by Mr-Darth** [comment]
> There is no entitytype to entitydata converter, this will break past compatability comparing an entity to an entitytype. Why did you remove entitytype?
>
> Not all entitydata match 1:1 with entitytype. EntityData is based off Entity classes, it's designed to be used a literal and return a created entity. Entitytype will be removed in the future as Skript doesn't use Spigot EntityType enum, just not at this current time, so support for it is required.

From the looks of it, `entitytype` was never meant to be supported (and it isn't, only the pattern accepted it). And EntityType is just EntityData with an amount.

File: src/main/java/ch/njol/skript/conditions/CondIsOfType.java [hunk 1]
```diff
@@ -38,42 +39,39 @@
...
+		return what.check(event,
+				(Checker<Object>) o1 -> types.check(event,
 						(Checker<Object>) o2 -> {
-							if (o2 instanceof ItemType && o1 instanceof ItemType) {
-								return ((ItemType) o2).isSupertypeOf((ItemType) o1);
+							if (o2 instanceof ItemType && o1 instanceof ItemStack) {
+								return ((ItemType) o2).isSupertypeOf(new ItemType((ItemStack) o1));
 							} else if (o2 instanceof EntityData && o1 instanceof Entity) {
 								return ((EntityData<?>) o2).isInstance((Entity) o1);
 							} else if (o2 instanceof ItemType && o1 instanceof Entity) {
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/conditions/CondIsOfType.java:77]: I know you didn't add it, but how exactly would an Entity and ItemType be comparable? Is this needed?

**Review by APickledWalrus — APPROVED** [review]
Might be worth looking into the EntityType thing as Lime mentioned

**Comment by TheLimeGlass** [comment]
@Mr-Darth

**Review by TheLimeGlass — DISMISSED** [review]
https://github.com/SkriptLang/Skript/pull/5073#issuecomment-1257034687

**Review by sovdeeth — CHANGES_REQUESTED** [review]
Some tests would be nice

**Comment by sovdeeth** [comment]
Do you still intend to work on this?

**Review by UnderscoreTud — APPROVED** [review]

File: src/test/skript/tests/syntaxes/conditions/CondIsOfType.sk [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    set {_armor-stand} to last spawned armor stand
+    assert {_armor-stand} is of type armor stand with "armor stand in a variable isn't of type armor stand"
+    delete last spawned armor stand
+
+    spawn enderpearl at spawn of "world"
+    assert last spawned enderpearl is of type ender pearl with "enderpearl isn't of type enderpearl"
+    assert last spawned enderpearl is not of type ghast with "enderpearl isn't of type ghast"
+    set {_enderpearl} to last spawned enderpearl
+    assert {_enderpearl} is of type ender pearl with "enderpearl in a variable isn't of type enderpearl"
+    delete last spawned enderpearl
```
Comment (sovdeeth) [inline src/test/skript/tests/syntaxes/conditions/CondIsOfType.sk:39]:
```suggestion
    delete last spawned enderpearl

```

**Review by sovdeeth — APPROVED** [review]

**Review by APickledWalrus — APPROVED** [review]

**Pull request merged**